app/src/main/java/org/mozilla/focus/fragment/WebFragment.java

```diff
@@ -104,7 +106,17 @@ public void applyLocale() {
     public void onPause() {
         final Session session = getSession();
         if (session != null) {
-            webView.saveWebViewState(session);
+            if (AppConstants.isGeckoBuild()) {
+                final CountDownLatch latch = new CountDownLatch(1);
+                webView.saveWebViewState(session, latch);
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    // Do nothing, the page was not saved
+                }
+            } else {
+                webView.saveWebViewState(session, null);
+            }
```

Comment:
As mentioned above: If you hide the CountDownLatch inside saveWebViewState() then this is just a single call here again.